docs/conf.py

```diff
@@ -202,3 +202,8 @@
      author, 'reana', 'One line description of project.',
      'Miscellaneous'),
 ]
+
+# Intersphinx configuration
```

Comment:
The intersphinx addition is nice :+1:  Two reflections:

I wonder whether we need `sphinx.ext.viewcode` rendering in the API doc chapter?  How likely it is that we (or casual visitors) are going to read the source code via RTFD? It does not cost anything to have this extension enabled... but if we don't use the feature than perhaps we can leave it out.... Dunno.  Anyway, we can keep it for now, if unsure. 

Another consideration is the doc chapter order: "Configuration" is an important section that should come after "Introduction", i.e. before "API".  The API reference section should come after any User guide or Developer guide we might have. Can you switch the order in `index.rst` as part of this PR?